internal/kafka/internal/services/kafka.go

```diff
@@ -1450,6 +1452,16 @@ func (k *kafkaService) ManagedKafkasRoutesTLSCertificate(kafkaRequest *dbapi.Kaf
 		return err
 	}
 
+	// check whether database update is needed
+	shouldUpdateCertInfoInTheDatabase = shouldUpdateCertInfoInTheDatabase ||
```

Comment:
Won't this return true if `shouldUpdateCertInfoInTheDatabase` is true independently of the values of KafkasRoutesBaseDomainTLSKeyRef, KafkasRoutesBaseDomainTLSCrtRef?

Reply:
Thanks, that's correct. I was about to answer that.